main.go

```diff
@@ -123,3 +145,12 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func getSecret(secretName string) []byte {
```

Comment:
I wonder whether it makes sense to update this kind of function, and instead of returning a byte array, we read from that known file location, decode it's contents, and attempt to build up a corev1.Secret object instead:

```go
func getSecretFromFile(secretName string) (*corev1.Secret, error) {
  data, err := ioutil.ReadFile(filepath.Join(secretBaseDir, secretName)
  if err != nil { return nil, err }
  
  secret := &corev1.Secret{
    ObjectMeta: metav1.ObjectMeta{
      ...
    }
    Data: map[string]struct{
      "password": ...
    }
    return secret, nil
}
```